### PR TITLE
Refactor category service and separate service and interface

### PR DESCRIPTION
**Pull request by ZinnaChoi**
Hi @JoshRyu 

I refactor the controller and service logic of the CATEGORY 
I also separate service and interface for all services
and also add the junit test for category to solve this [issue](https://github.com/JoshRyu/logistics/issues/34)

Plz check when you have time!! : )

### CODE REVIEW

File: backend/src/test/java/com/madeg/logistics/controller/CategoryControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    }
+
+    @Test
+    @WithMockUser(authorities = "ADMIN")
+    @Sql("/CategorySetup.sql")
+    @DisplayName("카테고리 삭제 실패 - 카테고리를 찾을 수 없음")
+    public void deleteCategoryFailCategoryNotFound() throws Exception {
+        TestUtil.performRequest(mockMvc, CATEGORY_API_URL + "/INVALID_CODE", null, "DELETE", 404, 404);
+    }
+
```
Comment (JoshRyu) [inline backend/src/test/java/com/madeg/logistics/controller/CategoryControllerTest.java:169]: can you add a following scenario to the test case?
- A is a parent category of B, C. User cannot delete A as long as there is a child category.

File: backend/src/test/java/com/madeg/logistics/controller/CategoryControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    @DisplayName("카테고리 생성 성공")
+    public void createCategorySuccess() throws Exception {
+        String requestBodyJson = objectMapper.writeValueAsString(new CategoryInput("TEST_CATEGORY", null, "양털이 몽글몽글"));
+        TestUtil.performRequest(mockMvc, CATEGORY_API_URL, requestBodyJson, "POST", 201, 201);
+    }
+
+    @Test
+    @WithMockUser(authorities = "ADMIN")
+    @Sql("/CategorySetup.sql")
+    @DisplayName("카테고리 생성 실패 - 존재하는 카테고리 이름")
```
Comment (JoshRyu) [inline backend/src/test/java/com/madeg/logistics/controller/CategoryControllerTest.java:60]: can you add test case for following scenario too?
- cateogory name cannot be null or empty string

**Review by JoshRyu — CHANGES_REQUESTED** [review]
Thanks for your contribution!
Nicely refactored, but I found few suggestions.
Please checkout my comments!

File: backend/src/test/java/com/madeg/logistics/controller/CategoryControllerTest.java (hunk 1 above)
Comment (ZinnaChoi) [reply backend/src/test/java/com/madeg/logistics/controller/CategoryControllerTest.java:178]: Oh i think i missed this case
thank you for your insight! 👍

**Review by JoshRyu — APPROVED** [review]
Looks great :)

**Pull request merged**